Applications/Spire/Source/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,116 @@
+#include "Spire/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.hpp"
+
+using namespace boost;
+using namespace boost::icl;
+using namespace boost::posix_time;
+using namespace Nexus;
+using namespace Spire;
+
+CachedOrderImbalanceIndicatorModel::CachedOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source)
+    : m_source_model(std::move(source)) {
+  auto [connection, promise] = m_source_model->subscribe(
```

Comment:
The ```CachedOrderImbalanceIndicatorModel``` must have ownership of the ```promise```, otherwise this will result in a dangling reference if the ```promise``` resolves after the ```CachedOrderImbalanceIndicatorModel``` dies.